app/src/test/java/tech/ula/utils/UserFeedbackUtilityTest.kt

```diff
@@ -0,0 +1,90 @@
+package tech.ula.utils
+
+import android.content.SharedPreferences
+import com.nhaarman.mockitokotlin2.whenever
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class UserFeedbackUtilityTest {
+
+    @Mock
+    lateinit var userFeedbackUtility: UserFeedbackUtility
+
+    @Mock
+    lateinit var sharedPrefs: SharedPreferences
+
+    private val minimumTimesAppOpenedToShowReview = 16
+
+    @Before
+    fun setup() {
+        userFeedbackUtility = UserFeedbackUtility(sharedPrefs)
+    }
+
+    @Test
+    fun `Asking for feedback is not appropriate if app opened for first time today`() {
+        setTimesAppOpened(minimumTimesAppOpenedToShowReview)
+        setUserGaveFeedback(false)
+        appOpenedNumberOfDaysAgo(0)
+
+        assertFalse(userFeedbackUtility.askingForFeedbackIsAppropriate())
+    }
+
+    @Test
+    fun `Asking for feedback is appropriate if app opened if app opened four days ago`() {
+        setTimesAppOpened(minimumTimesAppOpenedToShowReview)
+        setUserGaveFeedback(false)
+        appOpenedNumberOfDaysAgo(4)
+
+        assertTrue(userFeedbackUtility.askingForFeedbackIsAppropriate())
+    }
+
+    @Test
+    fun `Asking for feedback is not appropriate if app opened if app opened two days ago`() {
```

Comment:
Some repeated text in this function name.

Reply:
Good catch, I've removed the extra words